### PR TITLE
Add new monitoring pillar: Profiling

### PR DESCRIPTION
**Pull request by steeling**
Add a new monitoring pillar around profiling.

### CODE REVIEW

**Review by mtarng — APPROVED** [review]

File: observability/readme.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ Building observable systems enables development teams at CSE to measure how well
 1. [Logging](pillars/logging.md)
 2. [Tracing](pillars/tracing.md)
 3. [Metrics](pillars/metrics.md)
+4. [Profiling](pillars/profiling.md)
```
Comment (ivaramme) [inline observability/readme.md:29]: There's some really good content here. I would argue that across the industry (AFAIK) Profiling has not been mentioned so far as a pillar. I can definitely see the value of having this entry in the playbook, just not sure as a pillar of observability.  

Keen to hear others opinions...

**Review by ivaramme — CHANGES_REQUESTED** [review]

**Review by NarmathaBala — DISMISSED** [review]
The content looks great, pending opinions of the team about adding Profiling to 'Pillars'

File: observability/readme.md (hunk 1 above)
Comment (steeling) [reply observability/readme.md:29]: That's a fair point! I've used it pretty extensively in a past role as an SRE, when we had it enabled on all of our services as continuous profiling. It was a fantastic tool to correlate any crashes with load spikes, and then to see if there was any memory leak (especially for htose smaller leaks that might only show up under high load)


Anyways, happy to remove it as a pillar, and link to it elsewhere. I'll wait till  end of week to see if anybody else wants to offer an opinion in favor of keeping it here, before making that update.

[the remark this answers is not in the document]
Comment (ivaramme) [reply observability/readme.md:29]: Sorry @NarmathaBala - I swear I didn't see your comment last night before I added mine. I think both are aligned.

File: observability/pillars/profiling.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## Overview
+
+Profiling is a form of runtime analysis that measures various components of the runtime such as, memory allocation, garbage collection, threads and locks, callstacks, or frequency and duration of specific functions. It can be used to see which functions are the most costly in your binary, allowing you to focus your effort on removing the largest inefficiencies as quickly as possible. It can help you find deadlocks, memory leaks, or innefficient memory allocaiton, and help inform decisions around resource allocation (ie: CPU or RAM).
+
+## How to Profile your Applications
+
+Profiling is somewhat language dependent, so start off by searching for "profile $language" (some common tools are listed below). Additionally, Linux Perf is a good fallback, since a lot of languages have bindings in C/C++.
+
+Profiling does incur some cost, as it requires inspecting the callstack, and sometimes pausing the applicaiton alltogether (ie: to trigger a full GC in Java). It is recommended to continuously profile your services, say for 10s every 10 minutes. Consider the cost when deciding on tuning these parameters.
```
[the remark this answers is not in the document]
Comment (NarmathaBala) [reply observability/pillars/profiling.md:11]: @steeling  friendly ping to see if you are ready to make updates, I think pending these few items this is ready to me merged

File: observability/readme.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (steeling) [reply observability/readme.md]: kept it in that section, as I think people's eyes probably jump there first, but noted it's not typically considered a pillar. Let me know if you think it belongs in a separate section!

**Review by ivaramme — APPROVED** [review]

**Comment by fnocera** [comment]
Once #607 is merged in all outstanding PRs including this one will have to make a few changes and merge master to adopt the new MKDocs structure. This main change is that all the md files move into a `docs` folder. Please don't hesitate to reach out with any questions and a number of us are more than happy to help with this process - reach out to @fnocera @TessFerrandez @omri374 and @brockneedscoffee

**Pull request merged**